### PR TITLE
Update rootcoord state to unhealthy before stopping

### PR DESCRIPTION
**Pull request by congqixia**
See also #12721 

/kind improvement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -1180,7 +1180,7 @@ func (c *Core) Start() error {
 		Params.CreatedTime = time.Now()
 		Params.UpdatedTime = time.Now()
 
-		c.stateCode.Store(internalpb.StateCode_Healthy)
+		c.UpdateStateCode(internalpb.StateCode_Healthy)
```
Comment (jaime0815) [inline internal/rootcoord/root_coord.go:1183]: We'd better add a UI to cover this case.

Comment (congqixia) [reply internal/rootcoord/root_coord.go:1183]: Do you mean adding ut for `UpdateStateCode` function? Or add unit test to cover UpdateStateCode before performing any closing operations?

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -1189,9 +1189,10 @@ func (c *Core) Start() error {
 
 // Stop stop rootcoord
 func (c *Core) Stop() error {
+	c.UpdateStateCode(internalpb.StateCode_Abnormal)
```
Comment (jaime0815) [inline internal/rootcoord/root_coord.go:1192]: I mean add a ut to cover state has been updated before closing operations.

**Pull request merged**